src/dataregistry/DataRegistry.py

```diff
@@ -41,17 +46,49 @@ def __init__(
         verbose : bool
             True for more output.
         """
+
+        # Work out the location of the root directory
+        root_dir = self._get_root_dir(root_dir)
+
         # Establish connection to database
         db_connection = DbConnection(config_file, schema=schema, verbose=verbose)
 
         # Create registrar object
         self.Registrar = Registrar(
-            db_connection, owner=owner, owner_type=owner_type, root_dir=root_dir,
+            db_connection,
+            root_dir,
+            owner=owner,
+            owner_type=owner_type,
         )
 
         # Create query object
-        self.Query = Query(db_connection, root_dir=self.Registrar.root_dir)
+        self.Query = Query(db_connection, root_dir)
+
+    def _get_root_dir(self, root_dir):
+        """
+        What is the location of the root_dir we are pairing with?
+
+        Either this is a predefined "site", such as "NERSC", or a manually
```

Comment:
It's confusing for the same parameter to have two different interpretations (either site name or a file path).  If someone wanted to use "nersc" as a path (not a good idea probably, but also not inconceivable) they wouldn't be able to. 
There should be separate management of the site variable. When there is only one recognized site, it hardly matters, but when/if we have separate file areas for different sites, we probably need to allow people to set site either

- via a particular environment variable, e.g. DATAREGISTRY_SITE
- via a separate argument when creating a DataRegistry object.

I think priority should be

- if root_dir is explicitly specified, use that. Otherwise
- if site is explicitly specified, use that to look up corresponding root dir.  Otherwise
- if DATAREGISTRY_SITE has a value, use that to look up corresponding root dir
And if none of the above hold, I guess default to site == "nersc" and look up corresponding root dir

Reply:
Good suggestion.

I have reworked the function to work with a `root_dir` and a `site`, working with the priority order you have stated.

I've added a CI test to check the various pertubations.

I've added checks saying that Sqllite
- Cannot pass a `site`, must only pass `root_dir` or set `DATAREG_SITE`
- Cannot pass a `root_dir` that is the same as a `site` path in the site yaml file

For Sqlite I also had to update the CLI to accept `root_dir` and `site` as options (for the CI to work). Doesn't change anything by default however.